staging/src/k8s.io/client-go/tools/leaderelection/leaderelection.go

```diff
@@ -89,10 +89,13 @@ func NewLeaderElector(lec LeaderElectionConfig) (*LeaderElector, error) {
 	if lec.Lock == nil {
 		return nil, fmt.Errorf("Lock must not be nil.")
 	}
-	return &LeaderElector{
-		config: lec,
-		clock:  clock.RealClock{},
-	}, nil
+	le := LeaderElector{
+		config:  lec,
+		clock:   clock.RealClock{},
+		metrics: globalMetricsFactory.newLeaderMetrics(),
```

Comment:
This is really ugly.  What are we gaining by making prometheus metrics not the default?  Who is adding code level metrics plugins to k/k that we would approve?

This feels like a pointless abstraction - would like more clarity about why we are polluting the code with it.

Reply:
> This is really ugly. What are we gaining by making prometheus metrics not the default? Who is adding code level metrics plugins to k/k that we would approve?
> 
> This feels like a pointless abstraction - would like more clarity about why we are polluting the code with it.

Short answer: fewer deps from client-go.

Back when we started the project, prometheus wasn't as wide spread and we wanted to avoid the dependency. I think we should revisit that choice, but I would not choose to revisit it in this pull.

Reply:
As David says this is about reducing how many extra dependencies you have to pull in when you pull in client-go. The original version of the PR did not include the dependency breaking back flips but I was asked to add it in the standard manner, we do for other client-go metrics)